cloudify_agent/installer/__init__.py

```diff
@@ -353,7 +367,24 @@ def download(self, url, destination=None):
                 url,
                 output_path=destination,
                 skip_verification=not verify_cert,
-                certificate_file=local_cert_file)
+                certificate_file=self._get_certificate_file())
 
     def move(self, source, target):
         self.runner.move(source, target)
+
+    def _get_certificate_file(self):
+        cert_file = self.cloudify_agent['local_rest_cert_file']
+        cert_content = self.cloudify_agent.get('rest_cert_content')
+
+        if not cert_content:
```

Comment:
if the user has an image with the certificate already in place (or configured in userdata) then this call should not return `None`, no?